### PR TITLE
Upgrade path to Pydantic 2.0

### PR DESCRIPTION
**Pull request by turukawa**
Major path updates include:

- `frontend`:
  - Node 16 to 18
  - Latest Pinia requires changes in `stores`, where imports are not required (cause actual errors), and parameter declaration must happen in functions.
- `backend` and `celeryworker`:
  - Python 3.9 to 3.11
  - FastAPI 0.88 to 0.99 (Inboard 0.37 to 0.51)
  - Poetry to Hatch
  - Postgres 14 to 15

Review required on the new `hatch` `pyproject.toml` file, and to ensure that the environments are stable. @br3ndonland, I'd appreciate your review.

### CODE REVIEW

**Comment by turukawa** [comment]
The fix to `frontend` is related to a Vite error: https://github.com/scj2000/scj2000-web-v2/commit/b923143c106e352954771e19053d8eaddb414dbe

**Review by br3ndonland — CHANGES_REQUESTED** [review]
Thanks @turukawa. Great to see Hatch and inboard put to good use here. I have a few suggestions. Feel free to let me know if any further clarifications or changes are needed.

File: {{cookiecutter.project_slug}}/backend/backend.dockerfile [hunk 1]
```diff
@@ -1,16 +1,10 @@
-FROM ghcr.io/br3ndonland/inboard:fastapi-0.37.0-python3.9
-
-# Copy poetry.lock* in case it doesn't exist in the repo
-COPY ./app/pyproject.toml ./app/poetry.lock* /app/
+FROM ghcr.io/br3ndonland/inboard:fastapi-0.51.0-python3.11
 
+# Use file.name* in case it doesn't exist in the repo
+COPY ./app/pyproject.toml ./app/README.md ./app/__version__.py /app/
```
[the remark this answers is not in the document]
Comment (turukawa) [reply {{cookiecutter.project_slug}}/backend/backend.dockerfile:4]: Does this last line of the script achieve that? 
`COPY ./app/ /app/`
It won't be needed during build so it doesn't really matter where it happens, and this way nothing else above effects it until it's copied in last...

File: {{cookiecutter.project_slug}}/backend/backend.dockerfile [hunk 2]
```diff
@@ -1,16 +1,10 @@
...
-
-# Neomodel has shapely and libgeos as dependencies
-RUN apt-get update && apt-get install -y libgeos-dev
-
-# Allow installing dev dependencies to run tests
-ARG INSTALL_DEV=false
-RUN bash -c "if [ $INSTALL_DEV == 'true' ] ; then poetry install --no-interaction --no-root ; else poetry install --no-interaction --no-root --no-dev ; fi"
+ENV HATCH_ENV_TYPE_VIRTUAL_PATH=.venv
+RUN hatch env prune && hatch env create production
 RUN pip install --upgrade setuptools
```
[the remark this answers is not in the document]
Comment (turukawa) [reply {{cookiecutter.project_slug}}/backend/backend.dockerfile:8]: Ah, ok ... I was finding that something was removing `setuptools` and then causing hell to break lose when it's needed later. So this is just forcing it.

**Pull request merged**